src/code/z_player_lib.c

```diff
@@ -1580,8 +1650,8 @@ PlayerFaceIndices sPlayerFaces[] = {
 void Player_DrawImpl(PlayState* play, void** skeleton, Vec3s* jointTable, s32 dListCount, s32 lod,
                      PlayerTransformation playerForm, s32 boots, s32 face, OverrideLimbDrawFlex overrideLimbDraw,
                      PostLimbDrawFlex postLimbDraw, Actor* actor) {
-    s32 eyeIndex = (jointTable[22].x & 0xF) - 1;
-    s32 mouthIndex = ((jointTable[22].x >> 4) & 0xF) - 1;
+    s32 eyeIndex = (GET_FACE_FROM_JOINTTABLE(jointTable) & 0xF) - 1;
+    s32 mouthIndex = ((GET_FACE_FROM_JOINTTABLE(jointTable) >> 4) & 0xF) - 1;
```

Comment:
I wonder instead of 1 `GET_FACE_HAND_FROM_JOINTTABLE` macro, if you should make 4 separate macros instead:
`GET_EYE_INDEX_FROM_JOINTTABLE`
`GET_MOUTH_INDEX_FROM_JOINTTABLE`
`GET_LEFT_HAND_INDEX_FROM_JOINTTABLE`
`GET_RIGHT_HAND_INDEX_FROM_JOINTTABLE`

That way, you could include the masks and shifts within the macro

Reply:
> That way, you could include the masks and shifts within the macro

There's a small problem, the hand indices are handled a bit differently so the shifts can't be included in the macro itself:
```c

                s32 handIndex = GET_LEFT_HAND_INDEX_FROM_JOINTTABLE(player->skelAnime.jointTable);

                if (handIndex != 0) {
                    handIndex = (handIndex >> 12) - 1;
                    if (handIndex >= 2) {
                        handIndex = 0;
                    }
                    leftHandDLists = &D_801C095C[handIndex][D_801F59E0];
                }

// ...

                    s32 handIndex = GET_RIGHT_HAND_INDEX_FROM_JOINTTABLE(player->skelAnime.jointTable);

                    if (handIndex != 0) {
                        handIndex = (handIndex >> 8) - 1;
                        rightHandDLists = &D_801C0964[handIndex][D_801F59E0];
                    }
```